File: src/AdventureWorks.Application/DataEngine/Sales/SalesTaxRate/Queries/GetSalesTaxRates/SalesTaxRateLookupModel.cs
using System;
using AutoMapper;
using AdventureWorks.Application.Interfaces.Mapping;
using AdventureWorks.BaseDomain.Entities.Sales;
using System.Collections.Generic;
using AdventureWorks.Common.Extensions;
using AdventureWorks.BaseDomain.Enums;
using AdventureWorks.BaseDomain.Interfaces;
using AdventureWorks.BaseDomain.CustomTypes;
using AdventureWorks.Application.DataEngine.Person.StateProvince.Queries.GetStateProvinces;

namespace AdventureWorks.Application.DataEngine.Sales.SalesTaxRate.Queries.GetSalesTaxRates
{
    public partial class SalesTaxRateLookupModel : IHaveCustomMapping
    {
        public int SalesTaxRateID { get; set; }
        public int StateProvinceID { get; set; }
        public byte TaxType { get; set; }
        public decimal TaxRate { get; set; }
        public string Name { get; set; }
        public Guid Rowguid { get; set; }
        public DateTime ModifiedDate { get; set; }

        #region Navigation Properties

        // Uncomment the following line if you need child navigation properties (can cause a poor performance)
        /*  */
        // Uncomment the following line if you need parent navigation properties
        /*  */
        #endregion

        public void CreateMappings(Profile configuration)
        {
            configuration.CreateMap<Domain.Entities.Sales.SalesTaxRate, SalesTaxRateLookupModel>().IgnoreMissingDestinationMembers().CompensateWithDestinationValues();
            configuration.CreateMap<SalesTaxRateLookupModel, BaseSalesTaxRate>().IgnoreMissingDestinationMembers();
        }
    }
}

File: src/AdventureWorks.Web.Api/ClientApp/src/app/models/data/entities/sales/sales-tax-rate/sales-tax-rate-lookup-model.ts

export interface ISalesTaxRateLookupModel {
  salesTaxRateID: number;
  stateProvinceID: number;
  taxType: number;
  taxRate: number;
  name: string;
  rowguid: string;
  modifiedDate: Date | string;

  // Uncomment the following line if you need child navigation properties (can cause a poor performance)
  /*  */
  // Uncomment the following line if you need parent navigation properties
  /*  */
}

export class SalesTaxRateLookupModel implements ISalesTaxRateLookupModel {
  salesTaxRateID: number;
  stateProvinceID: number;
  taxType: number;
  taxRate: number;
  name: string;
  rowguid: string;
  modifiedDate: Date | string;

  // Uncomment the following line if you need child navigation properties (can cause a poor performance)
  /*  */
  // Uncomment the following line if you need parent navigation properties
  /*  */

  constructor(init?: Partial<SalesTaxRateLookupModel>) {
    if (!!init) {
      Object.assign<SalesTaxRateLookupModel, Partial<SalesTaxRateLookupModel>>(this, init);
    }
  }
}
